diff -up gnome-commander-1.16.0/src/tags/gnome-cmd-tags-exiv2.cc.omv~ gnome-commander-1.16.0/src/tags/gnome-cmd-tags-exiv2.cc
--- gnome-commander-1.16.0/src/tags/gnome-cmd-tags-exiv2.cc.omv~	2023-05-18 05:58:32.814304164 +0200
+++ gnome-commander-1.16.0/src/tags/gnome-cmd-tags-exiv2.cc	2023-05-18 05:58:41.258386649 +0200
@@ -398,7 +398,7 @@ void gcmd_tags_exiv2_load_metadata(Gnome
 #ifdef HAVE_EXIV2
     try
     {
-        Image::AutoPtr image = ImageFactory::open(fname);
+        Image::UniquePtr image = ImageFactory::open(fname);
 
         image->readMetadata();
 
@@ -406,7 +406,7 @@ void gcmd_tags_exiv2_load_metadata(Gnome
         readTags(f->metadata, image->iptcData());
     }
 
-    catch (AnyError &e)
+    catch (Error &e)
     {
     }
 #endif
