diff -p -up gnome-commander-1.2.8.5/src/gnome-cmd-prepare-xfer-dialog.cc.fmt-str gnome-commander-1.2.8.5/src/gnome-cmd-prepare-xfer-dialog.cc
--- gnome-commander-1.2.8.5/src/gnome-cmd-prepare-xfer-dialog.cc.fmt-str	2010-02-13 21:39:25.000000000 +0100
+++ gnome-commander-1.2.8.5/src/gnome-cmd-prepare-xfer-dialog.cc	2010-02-14 12:02:29.000000000 +0100
@@ -192,6 +192,7 @@ static void on_ok (GtkButton *button, Gn
                                                             (GtkDialogFlags) 0,
                                                             GTK_MESSAGE_QUESTION,
                                                             GTK_BUTTONS_OK_CANCEL,
+							    "%s",
                                                             msg);
                 gint choice = gtk_dialog_run (GTK_DIALOG (dialog));
                 gtk_widget_destroy (dialog);
diff -p -up gnome-commander-1.2.8.5/src/gnome-cmd-xml-config.cc.fmt-str gnome-commander-1.2.8.5/src/gnome-cmd-xml-config.cc
--- gnome-commander-1.2.8.5/src/gnome-cmd-xml-config.cc.fmt-str	2010-02-13 21:39:25.000000000 +0100
+++ gnome-commander-1.2.8.5/src/gnome-cmd-xml-config.cc	2010-02-14 12:02:29.000000000 +0100
@@ -546,7 +546,7 @@ gboolean gnome_cmd_xml_config_parse (con
     if (!g_markup_parse_context_parse (context, xml, xml_len, &error) ||
         !g_markup_parse_context_end_parse (context, &error))
     {
-        g_warning (error->message);
+      g_warning ("%s", error->message);
         g_error_free (error);
     }
 
@@ -567,7 +567,7 @@ gboolean gnome_cmd_xml_config_load (cons
 
     if (!g_file_get_contents (path, &xml, &xml_len, &error))
     {
-        g_warning (error->message);
+      g_warning ("%s", error->message);
         g_error_free (error);
 
         return FALSE;
diff -p -up gnome-commander-1.2.8.5/src/utils.cc.fmt-str gnome-commander-1.2.8.5/src/utils.cc
--- gnome-commander-1.2.8.5/src/utils.cc.fmt-str	2010-02-13 21:39:25.000000000 +0100
+++ gnome-commander-1.2.8.5/src/utils.cc	2010-02-14 12:02:29.000000000 +0100
@@ -202,7 +202,7 @@ gint run_simple_dialog (GtkWidget *paren
     button_titles = convert_varargs_to_name_array (button_title_args);
     va_end (button_title_args);
 
-    dialog = gtk_message_dialog_new (GTK_WINDOW (main_win), GTK_DIALOG_MODAL, msg_type, GTK_BUTTONS_NONE, text);
+    dialog = gtk_message_dialog_new (GTK_WINDOW (main_win), GTK_DIALOG_MODAL, msg_type, GTK_BUTTONS_NONE, "%s", text);
     if (title)
         gtk_window_set_title (GTK_WINDOW (dialog), title);
 
@@ -599,7 +599,7 @@ void mime_exec_single (GnomeCmdFile *f)
         {
             gchar *msg = g_strdup_printf (_("%s does not know how to open remote file. Do you want to download the file to a temporary location and then open it?"), gnome_cmd_app_get_name (app));
             GtkWidget *dialog = gtk_message_dialog_new (GTK_WINDOW (main_win), GTK_DIALOG_MODAL,
-                                                        GTK_MESSAGE_QUESTION, GTK_BUTTONS_YES_NO, msg);
+                                                        GTK_MESSAGE_QUESTION, GTK_BUTTONS_YES_NO, "%s", msg);
             TmpDlData *dldata = g_new0 (TmpDlData, 1);
             args[0] = (gpointer) app;
             // args[2] is NULL here (don't set exec dir for temporarily downloaded files)
diff -p -up gnome-commander-1.2.8.5/src/utils.h.fmt-str gnome-commander-1.2.8.5/src/utils.h
--- gnome-commander-1.2.8.5/src/utils.h.fmt-str	2010-02-13 21:39:25.000000000 +0100
+++ gnome-commander-1.2.8.5/src/utils.h	2010-02-14 12:02:29.000000000 +0100
@@ -238,10 +238,11 @@ inline void gnome_cmd_show_message (GtkW
                                              GTK_DIALOG_DESTROY_WITH_PARENT,
                                              GTK_MESSAGE_ERROR,
                                              GTK_BUTTONS_OK,
+					     "%s",
                                              message.c_str());
 
     if (secondary_text)
-        gtk_message_dialog_format_secondary_text (GTK_MESSAGE_DIALOG (dlg), secondary_text);
+        gtk_message_dialog_format_secondary_text (GTK_MESSAGE_DIALOG (dlg), "%s", secondary_text);
 
     gtk_dialog_run (GTK_DIALOG (dlg));
     gtk_widget_destroy (dlg);
diff -p -up gnome-commander-1.2.8.5/libgcmd/libgcmd-widget-factory.cc~ gnome-commander-1.2.8.5/libgcmd/libgcmd-widget-factory.cc
--- gnome-commander-1.2.8.5/libgcmd/libgcmd-widget-factory.cc~	2010-02-13 21:39:25.000000000 +0100
+++ gnome-commander-1.2.8.5/libgcmd/libgcmd-widget-factory.cc	2010-02-14 12:40:23.000000000 +0100
@@ -652,7 +652,7 @@ create_error_dialog (const gchar *msg, .
     va_end (argptr);
 
     dialog = gtk_message_dialog_new (GTK_WINDOW (main_win_widget), GTK_DIALOG_MODAL,
-                                     GTK_MESSAGE_ERROR, GTK_BUTTONS_OK, string);
+                                     GTK_MESSAGE_ERROR, GTK_BUTTONS_OK, "%s", string);
 
     gtk_signal_connect (GTK_OBJECT (dialog), "response", GTK_SIGNAL_FUNC (on_response), dialog);
 
@@ -673,7 +673,7 @@ create_warning_dialog (const gchar *msg,
 
     dialog = gtk_message_dialog_new (
         GTK_WINDOW (main_win_widget), GTK_DIALOG_MODAL,
-        GTK_MESSAGE_WARNING, GTK_BUTTONS_OK, msg);
+        GTK_MESSAGE_WARNING, GTK_BUTTONS_OK, "%s", msg);
 
     gtk_signal_connect (GTK_OBJECT (dialog), "response",
                         GTK_SIGNAL_FUNC (on_response), dialog);
